presto-hive/src/main/java/com/facebook/presto/hive/statistics/ParquetQuickStatsBuilder.java

```diff
@@ -0,0 +1,344 @@
+                        FloatStatistics asFloatStats = ((FloatStatistics) colStats);
+                        toMerge.setMinValue(asFloatStats.getMin());
+                        toMerge.setMaxValue(asFloatStats.getMax());
+                        toMerge.addToNullsCount(nullsCount);
+                        toMerge.addToRowCount(rowCount);
+                        rolledUpColStats.put(columnKey, toMerge);
+                        break;
+                    }
+                    case BOOLEAN: {
+                        ColumnQuickStats<Boolean> toMerge = (ColumnQuickStats<Boolean>) rolledUpColStats.getOrDefault(columnKey, new ColumnQuickStats<>(columnName, Boolean.class));
+                        toMerge.addToNullsCount(nullsCount);
+                        toMerge.addToRowCount(rowCount);
+                        // TODO : Boolean stats store trueCount and falseCount
+                        rolledUpColStats.put(columnKey, toMerge);
+                        break;
+                    }
+                    case DATE: {
+                        ColumnQuickStats<ChronoLocalDate> toMerge = (ColumnQuickStats<ChronoLocalDate>) rolledUpColStats.getOrDefault(columnKey,
+                                new ColumnQuickStats<>(columnName, ChronoLocalDate.class));
+                        IntStatistics asIntStats = ((IntStatistics) colStats);
+                        toMerge.setMinValue(LocalDate.ofEpochDay(asIntStats.getMin()));
+                        toMerge.setMaxValue(LocalDate.ofEpochDay(asIntStats.getMax()));
+                        toMerge.addToNullsCount(nullsCount);
+                        toMerge.addToRowCount(rowCount);
+                        rolledUpColStats.put(columnKey, toMerge);
+                        break;
+                    }
+                    default:
+                    case SLICE: {
+                        ColumnQuickStats<Slice> toMerge = (ColumnQuickStats<Slice>) rolledUpColStats.getOrDefault(columnKey, new ColumnQuickStats<>(columnName, Slice.class));
+                        toMerge.addToNullsCount(nullsCount);
+                        toMerge.addToRowCount(rowCount);
+                        rolledUpColStats.put(columnKey, toMerge);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public PartitionQuickStats buildQuickStats(ConnectorSession session, SemiTransactionalHiveMetastore metastore,
+            SchemaTableName table, MetastoreContext metastoreContext, String partitionId, Iterator<HiveFileInfo> files)
+    {
+        requireNonNull(session);
+        requireNonNull(metastore);
+        requireNonNull(table);
+        requireNonNull(metastoreContext);
+        requireNonNull(partitionId);
+        requireNonNull(files);
+
+        if (!files.hasNext()) {
+            return PartitionQuickStats.EMPTY;
+        }
+
+        StorageFormat storageFormat;
```

Comment:
This logic in determining storageFormat would have to be replicated for each <file format>QuickStatsBuilder. Would it make sense to move it into QuickStatsBuilder and then pick the right strategy based on the storageFormat there?

Reply:
Yes, leaving it for a future refactor. Will add a note for this.